snapshot/snapshotfs/upload.go

```diff
@@ -779,8 +784,36 @@ func (u *Uploader) Upload(
 	}
 
 	s.IncompleteReason = u.cancelReason()
-	s.EndTime = u.repo.Time()
+	endTime := u.repo.Time()
 	s.Stats = u.stats
 
+	s.StartTime = startTime
+	s.EndTime = endTime
+
+	u.overrideTimestamps(s, startTime, endTime)
+
 	return s, nil
 }
+
+func (u *Uploader) overrideTimestamps(
```

Comment:
can this logic be outside of the uploader? original idea was for uploader to potentially be shared between multiple snapshots and and having shared overrides does not work.

you don't really need "u" in this function - just the two overrides that can be stored in the "cli" package.

Reply:
btw, other than this the PR looks very good.

Reply:
Yep that makes sense - I was trying to avoid creating new objects or passing values down the chain but I think I worked out a good way to do it